CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+# v0.1.2
+
+- Added a safety check for enforcing that the VersionMaps provided for
+  (de)serialization are up-to-date with the latest versions of the
+  structs/enums/unions.
```

Comment:
Sorry for insisting on this :(, I think the enum deserves a separate entry in which we mention that only for serialization we're doing that enforcement. Better long than inaccurate or wrong (unless I'm missing something here, in which case feel free to disregard my comment).
Also, side note: should we log an issue for the enum thing?

Reply:
I think we should keep change log entries short and move details into the docs.

Reply:
I agree that changelog entries should be rather short. I'll log an issue to keep in mind that we don't do versioned deserialization for enums yet.

Reply:
Sorry, for me it's wrong, won't block the PR, but can't approve.

Reply:
I am unsure why you think this is wrong, a check on the serialization path is enough, duplicating the check on deserialization should be nice but not mandatory. The integration test will test both serialiazation/deserialization. More than that , iirc there is no need to perform versioned deserialization for enums as we disallow removal of enum variants until explicit variant discriminators can be used with data tuples.

Reply:
Synced offline with @sandreim and it looks like I didn't make myself enough clear :(.
I'll try again :)), because we don't support deserialization for enums, it's inaccurate/imprecise (if wrong is too much) to say we added a check for (**de**)serialization .... enums. The idea here was to somehow rephrase or split it in two in order to avoid possible confusions (the CHANGELOG should aim to be as clear as possible IMO).